### PR TITLE
fix: Return NomComs, not Groups, from active_nomcoms filter

### LINKED ISSUE #5723: Don't crash when a nomcom is partially set up.
Right now, if a nomcom group is created, and the associated NomCom object is not, anyone with a role in the nomcom group cannot use the datatracker - the construction of the menu crashes.

The places that crash need to be protected against this misconfiguration (especially while the configuration of the Group and NomCom objects are done manually).

See also https://github.com/ietf-tools/datatracker/issues/3289

### PR DESCRIPTION
**Pull request by jennifer-richards**
Fixes #5723

### CODE REVIEW

File: ietf/group/templatetags/group_filters.py [hunk 1]
```diff
@@ -19,14 +19,13 @@ def active_nomcoms(user):
...
-        type_id='nomcom',
-        state__slug='active').distinct().select_related("type"))
-
-    return groups
+    return list(
+        NomCom.objects.filter(
+            group__role__person__user=user,
+            group__type_id='nomcom',  # just in case...
+            group__state__slug='active',
+        )
```
Comment (rjsparks) [inline ietf/group/templatetags/group_filters.py:27]: probably want distinct() here - several people end up with multiple roles in a NomCom.group.

Comment (rjsparks) [reply ietf/group/templatetags/group_filters.py:27]: also consider an order_by, perhaps on `group__acronym` (but maybe making the implicit `pk` ordering explicit is enough.

Comment (rjsparks) [reply ietf/group/templatetags/group_filters.py:27]: Where was the `select_related("type")` used in the existing code?

**Review by rjsparks — CHANGES_REQUESTED** [review]

File: ietf/group/templatetags/group_filters.py (hunk 1 above)
Comment (rjsparks) [reply ietf/group/templatetags/group_filters.py:27]: (looks like it _wasn't_. Some other code must have gone away...)

Comment (jennifer-richards) [reply ietf/group/templatetags/group_filters.py:27]: That's right, it wasn't being used. Will add the `distinct()` and an ordering

**Review by rjsparks — APPROVED** [review]

**Pull request merged**